### PR TITLE
Adjust accuracy tolerance in beta_inc_inv to accommodate Float64

### PR DESCRIPTION
**Pull request by andreasnoack**
Also, adjust the tolerance when testing `beta_inc_inv` and add a new test. One of the existing test cases failed with the adjusted test tolerance before the change of this PR. See the comment in the source some more details on the change here.

Fixes https://github.com/JuliaMath/SpecialFunctions.jl/issues/375

### CODE REVIEW

File: src/beta_inc.jl [hunk 1]
```diff
@@ -971,7 +972,29 @@ function _beta_inc_inv(a::Float64, b::Float64, p::Float64, q::Float64=1-p)
...
+    # This first argument was proposed in
+    #
+    # K. J. Berry, P. W. Mielke, Jr and G. W. Cran (1990).
+    # Algorithm AS R83: A Remark on Algorithm AS 109: Inverse of the
+    #   Incomplete Beta Function Ratio.
+    # Journal of the Royal Statistical Society.
+    # Series C (Applied Statistics), 39(2), 309–310. doi:10.2307/2347779
+    #
+    # but the last term has been changed from 13 to 34 since the
+    # the original article
```
Comment (devmotion) [inline src/beta_inc.jl:984]:
```suggestion
    # original article
```

File: src/beta_inc.jl [hunk 2]
```diff
@@ -930,7 +930,8 @@ function _beta_inc_inv(a::Float64, b::Float64, p::Float64, q::Float64=1-p)
     x = p
     r = sqrt(-2*log(p))
     p_approx = r - @horner(r, 2.30753e+00, 0.27061e+00) / @horner(r, 1.0, .99229e+00, .04481e+00)
-    fpu = 1e-30
+    sae = log10(floatmin(Float64))
+    fpu = 10^sae
```
Comment (stevengj) [inline src/beta_inc.jl:934]:
```suggestion
    fpu = exp10(sae)
```
with this change, it looks like LLVM can constant-fold the calculation of `fpu`.

Comment (simonbyrne) [inline src/beta_inc.jl:934]: Isn't this just
```suggestion
    fpu = floatmin(Float64)
```

Comment (andreasnoack) [reply src/beta_inc.jl:934]: `sae` is used elsewhere but indeed `fpu` should just be computed with `floatmin`.

**Comment by andreasnoack** [comment]
Bump. This one is needed over at https://github.com/JuliaStats/StatsFuns.jl/pull/113 so it would be great to get merged.

**Pull request merged**